enhancements/cvo/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,142 @@
+---
+title: use-manifest-annotation-for-object-removal
+authors:
+  - "@jottofar"
+reviewers:
+  - "@LalatenduMohanty"
+  - "@wking"
+  - "@crawford"
+  - "@sdodson"
+approvers:
+  - "@crawford"
+  - "@sdodson"
+creation-date: 2020-08-03
+last-updated: 2020-08-04
+status: implementable
+---
+
+# Use Manifest Annotation For Object Removal
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently CVO managed object removal is handled by jobs.
+Jobs have many immutable components, so they aren't a great match for CVO managed manifests.
+Instead of the current approach manifests will be used by adding a new annotation to a manifest which will be handled by the CVO causing it to delete the in-cluster object rather than creating/reconciling the manifest.
+This will provide a more straightforward way to remove content.
+
+## Motivation
+
+There should be a straightforward way to remove CVO managed objects.
+
+### Goals
+
+* Users will be able to remove any of the currently managed CVO objects by simply modifying the object's exisitng manifest and adding the new delete annotation.
+
+### Non-Goals
+
+* This enhancement does not add any new objects for managment by CVO therefore the new delete annotation only applies to the currently supported manifests and any that may be added in the future.
+
+## Proposal
+
+When the following annotation appears in a CVO supported manifest and is set to "true" the associated object will be removed from the cluster by the CVO:
+
+```
+apiVersion: apps/v1
+...
+metadata:
+...
+  annotations:
+    release.openshift.io/delete: "true" 
+```
+
+### User Stories
+
+#### Story 1
+
+Remove the cluster-autoscaler-operator deployment.
+The existing cluster-autoscaler-operator deployment manifest 0000_50_cluster-autoscaler-operator_07_deployment.yaml is modified to contain the delete annotation:
+```
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler-operator
+  namespace: openshift-machine-api
+  labels:
+    k8s-app: cluster-autoscaler-operator
+  annotations:
+    exclude.release.openshift.io/internal-openshift-hosted: "true"
```

Comment:
`exclude.release.openshift.io/...` has never been formally defined.  I'd rather remove it here, because there's no harm in including a for-delete manifest in the manifest list, even if the for-inclusion manifests was never part of the list for a given target.

Reply:
Changed in commit [262765d](https://github.com/openshift/enhancements/pull/419/commits/262765d0e846f11bb32d17fe4841e5b9917d4a82)